client/inventory/client.go

```diff
@@ -84,6 +94,7 @@ func (c *client) CheckHealth(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	defer rsp.Body.Close()
```

Comment:
I've seen these in the other PRs: the missing differed call to Close is indeed a bug, well done.